### PR TITLE
Added documentation files

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**
- Contains README.md file, which contains sub-links to the docs folder.
- The docs folder contains the following files:
        - INSTALLATION.md
        - ENV.md
        - DEPLOYMENT.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,13 @@
-# auth-layer
\ No newline at end of file
+[![License: MIT](https://img.shields.io/github/license/avantifellows/auth-layer?color=blue&style=flat-square)](LICENSE)
+[![Discord](https://img.shields.io/discord/717975833226248303.svg?label=&logo=discord&logoColor=ffffff&color=7389D8&labelColor=6A7EC2&style=flat-square)](https://discord.gg/29qYD7fZtZ)
+
+# Authentication Layer
+This layer is being built to authenticate and authorize users to use any reource at Avanti.
```
Comment (deepansh96) [inline README.md]:
```suggestion
This layer is being built to authenticate, authorize and keep track of the students who will use Avanti Fellows resources or any external resource.
```

File: README.md [hunk 2]
```diff
@@ -1 +1,13 @@
-# auth-layer
\ No newline at end of file
+[![License: MIT](https://img.shields.io/github/license/avantifellows/auth-layer?color=blue&style=flat-square)](LICENSE)
+[![Discord](https://img.shields.io/discord/717975833226248303.svg?label=&logo=discord&logoColor=ffffff&color=7389D8&labelColor=6A7EC2&style=flat-square)](https://discord.gg/29qYD7fZtZ)
+
+# Authentication Layer
+This layer is being built to authenticate and authorize users to use any reource at Avanti. 
+This repo contains code for the frontned in VueJS.
```
Comment (deepansh96) [inline README.md]:
```suggestion
This repo contains code for the frontend of the layer written in VueJS.
```

File: docs/DEPLOYMENT.MD [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# DEPLOYMENT
+This document covers steps on setting up this repository on various cloud hosting providers.
+
+# AWS
+
+# Staging
+1. Login to your aws console.
+2. Create a S3 bucket and uncheck `Block all plublic access`.
+3. Enable static website hosting and the bucket endpoint is what will be used to set up Cloudfront distribution.
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]:
```suggestion
3. Enable static website hosting and note down the bucket endpoint. We'll use it to set up a Cloudfront distribution later.
```

File: docs/DEPLOYMENT.MD [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+4. In Cloudfront, create a distribution with the S3 origin, created in step 4.
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]:
```suggestion
4. In Cloudfront, create a distribution with the S3 origin, created in the step above.
```

File: docs/DEPLOYMENT.MD [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+5. If you choose to set up a SSL, do so at this step.
+6. Use a DNS provider of your choice. The target address would be the address Cloudfront has generated for you.
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]:
```suggestion
6. Use a DNS provider of your choice. Give an appropriate CNAME and enter the target address. This would be the address of the newly created Cloudfront distribution.
```

File: docs/DEPLOYMENT.MD [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+7. Next, go to your GitHub repository and create a new environment from settings tab.
+        i. Name the environment as Staging.
+        ii. Make sure you have added the following GitHub secrets on repository level. If not, add these as your environment secrets.
+                AWS_ACCESS_KEY_ID
+                AWS_SECRET_ACCESS_KEY
+                AWS_REGION
+        iii. Additionally, refer to .env.example file to set all the other required environment variables for your project.
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]: Why is this required?

File: docs/DEPLOYMENT.MD [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+8. To deploy your code on to staging, use `npm deploy-staging/yarn deploy-staging`.
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]:
```suggestion
8. To deploy your code on to staging, use `npm run deploy-staging` or `yarn deploy-staging`.
```
`npm deploy-staging` is not the right command

File: docs/DEPLOYMENT.MD [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD:19]: Also, I think you missed the steps of setting up a script inside `package.json`

File: docs/INSTALLATION.MD [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# Installation steps
+Follow the steps below to set up the repo for development.
+1. Clone the repository and change the working directory
+    ```sh
+    git clone https://github.com/avantifellows/auth-layer.git
+    cd plio-frontend
+    ```
+    
+2. Set up your `.env` file by copying `.env.example` file
```
Comment (deepansh96) [inline docs/INSTALLATION.MD:9]: Just also mention here the part of setting up .env files for staging and prod
`.env.staging` and `.env.production`

File: docs/INSTALLATION.MD [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    ```sh
+    cp .env.example .env
+    ```
+
+3. Update environment variables in your `.env` file based on your environment. For all available settings, see our [Environment variables guide](ENV.md).
+
+4. Your frontend app should be accessible at http://localhost:8080.
```
Comment (deepansh96) [inline docs/INSTALLATION.MD]: You forgot to mention how to serve the app 
`npm run serve`

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: docs/DEPLOYMENT.MD (hunk 6 above)
Comment (sudheshna-donthineni) [reply docs/DEPLOYMENT.MD]: You'll need to add AWS credentials, no? I saw it in the plio readme too. not required?

Comment (deepansh96) [reply docs/DEPLOYMENT.MD]: No no
AWS credentials are required to push the file on s3. But you do that locally using `yarn deploy`. So you need those creds locally, not stored on github.

These are only required when github is pushing the file to s3 on our behalf. And that is in the case of CD. So it's not really required here. let me know if it's not clear

Comment (sudheshna-donthineni) [reply docs/DEPLOYMENT.MD]: Oh! Understood, makes sense!

File: docs/DEPLOYMENT.MD [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+        i. Name the environment as Staging.
+        ii. Make sure you have added the following GitHub secrets on repository level. If not, add these as your environment secrets.
+                AWS_ACCESS_KEY_ID
+                AWS_SECRET_ACCESS_KEY
+                AWS_REGION
+        iii. Additionally, refer to .env.example file to set all the other required environment variables for your project.
+
+8. Add the following lines to the `scripts` object in `package.json`.
+    - "deploy-staging": "vue-cli-service build --mode staging && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGION s3 sync ./dist $S3_STAGING_BUCKET_NAME --delete --acl public-read",
+    "deploy": "vue-cli-service build && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGION s3 sync ./dist $S3_PROD_BUCKET_NAME --delete --acl public-read"
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]: Screenshot 2021-08-05 at 4 01 00 PM
This looks like as shown in the screenshot in markdown
Better way to write this would be using code blocks in markdown
See this https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

File: docs/DEPLOYMENT.MD [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+1. Login to your aws console.
+2. Create a S3 bucket and uncheck `Block all plublic access`.
+3. Enable static website hosting and note down the bucket endpoint. We'll use it to set up a Cloudfront distribution later.
+4. In Cloudfront, create a distribution with the S3 origin, created in the step above.
+5. If you choose to set up a SSL, do so at this step.
+6. Use a DNS provider of your choice. Give an appropriate CNAME and enter the target address. This would be the address of the newly created Cloudfront distribution.
+7. Add the following lines to the `scripts` object in `package.json`.
+    
+    ```"deploy-staging": "vue-cli-service build --mode staging && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGION s3 sync ./dist $S3_STAGING_BUCKET_NAME --delete --acl public-read", ```
+    ```"deploy": "vue-cli-service build && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGION s3 sync ./dist $S3_PROD_BUCKET_NAME --delete --acl public-read"```
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]: Screenshot 2021-08-05 at 4 12 11 PM
could be made better
Use this

```json
{
    "deploy-staging": "vue-cli-service build --mode staging && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGION s3 sync ./dist s3://$S3_STAGING_BUCKET_NAME --delete --acl public-read",
     "deploy": "vue-cli-service build && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGIO s3 sync ./dist s3://$S3_STAGING_BUCKET_NAME --delete --acl public-read"
}
```

**Review by deepansh96 — APPROVED** [review]
